src/main/java/HttpHandlerSelector.java

```diff
@@ -1,114 +1,24 @@
 package com.td.HttpServer;
-import java.util.*;
-import java.text.DateFormat;
-import java.text.SimpleDateFormat;
-import java.io.IOException;
 
-// TODO This class is too big, and has multiple responsibilities.
-//      It needs to be broken up into multiple classs.
-//      Two classes that could be pulled out of this are
-//         HttpResponseBuilder & HeaderBuilder
+import java.util.*;
 
 public class HttpHandlerSelector {
 
-  IFileIO fileIO;
-  HashMap<String, String> standardHeaders;
-
-  private final String CODE_200 = "HTTP/1.1 200 OK";
-  private final String CODE_404 = "HTTP/1.1 404 Not Found";
-  private final String CODE_400 = "HTTP/1.1 400 Bad Request";
-
-  private final String HTML = ".html";
+  HttpHandleRequest getRequest;
 
-  public HttpHandlerSelector(IFileIO fileIO) {
-    this.fileIO = fileIO;
-    standardHeaders = new HashMap<String, String>();
-    standardHeaders.put("Server", "Simple Http Server written in Java");
+  public HttpHandlerSelector(HttpHandleRequest getRequest) {
```

Comment:
So, if `HttpHandleRequest` is an abstract class, presumably you could be passing in any kind of request handler here, right? Not necessarily a GET request handler? Seems like you'll want a more generic name for the variable getting passed in (`requestHandler`?). Also, shouldn't this approach make the `if` check on `request.method()` below unnecessary?